gdpr/basic_enforcement.go

```diff
@@ -0,0 +1,42 @@
+package gdpr
+
+import (
+	tcf2 "github.com/prebid/go-gdpr/vendorconsent/tcf2"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// BasicEnforcement determines if legal basis is satisfied for a given purpose and bidder using
+// the TCF2 basic enforcement algorithm. The algorithm is a high-level mode of consent confirmation
+// that looks for a good-faith indication that the user has provided consent or legal basis signals
+// necessary to perform a privacy-protected activity. The algorithm does not involve the GVL.
+// BasicEnforcement implements the PurposeEnforcer interface
+type BasicEnforcement struct {
+	cfg purposeConfig
+}
+
+// LegalBasis determines if legal basis is satisfied for a given purpose and bidder based on user consent
+// and legal basis signals.
+func (be *BasicEnforcement) LegalBasis(vendorInfo VendorInfo, bidder openrtb_ext.BidderName, consent tcf2.ConsentMetadata, overrides Overrides) bool {
+	if !be.cfg.EnforcePurpose && !be.cfg.EnforceVendors {
+		return true
+	}
+	if be.cfg.vendorException(bidder) && !overrides.blockVendorExceptions {
+		return true
+	}
+	if !be.cfg.EnforcePurpose && be.cfg.basicEnforcementVendor(bidder) {
+		return true
+	}
+	if be.cfg.EnforcePurpose && consent.PurposeAllowed(be.cfg.PurposeID) && be.cfg.basicEnforcementVendor(bidder) {
+		return true
+	}
+	if be.cfg.EnforcePurpose && consent.PurposeLITransparency(be.cfg.PurposeID) && overrides.allowLITransparency {
+		return true
+	}
+	if be.cfg.EnforcePurpose && !consent.PurposeAllowed(be.cfg.PurposeID) {
+		return false
+	}
+	if !be.cfg.EnforceVendors {
+		return true
+	}
+	return consent.VendorConsent(vendorInfo.vendorID)
```

Comment:
To clarify, is this section of if statements, a representation of those truth tables you showed me?

Reply:
Yes that's correct. The requirements document contains truth tables for both basic and full enforcement. The `LegalBasis` method in `basic_enforcement.go` and `full_enforcement.go` implement those truth tables.